schema/commonmeta/commonmeta.schema.json

```diff
@@ -0,0 +1,541 @@
+            "description": "The contributors to the resource.",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "id": {
+                        "description": "The unique identifier for the contributor.",
+                        "type": "string"
+                    },
+                    "type": {
+                        "description": "The type of the contributor.",
+                        "type": "string",
+                        "enum": [
+                            "Organization",
+                            "Person"
+                        ]
+                    },
+                    "name": {
+                        "description": "The name of the contributor.",
+                        "type": "string"
+                    },
+                    "givenName": {
+                        "description": "The given name of the contributor.",
+                        "type": "string"
+                    },
+                    "familyName": {
+                        "description": "The family name of the contributor.",
+                        "type": "string"
+                    },
+                    "affiliation": {
+                        "$ref": "#/definitions/affiliations"
+                    }
+                }
+            }
+        },
+        "subjects": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "subject": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "subject"
+                ]
+            },
+            "uniqueItems": true
+        },
+        "sizes": {
+            "description": "The sizes of the resource.",
+            "type": "array",
+            "items": {
+                "type": "string"
+            },
+            "uniqueItems": true
+        },
+        "formats": {
+            "description": "The formats of the resource.",
+            "type": "array",
+            "items": {
+                "type": "string"
+            },
+            "uniqueItems": true
+        },
+        "language": {
+            "description": "The language of the resource. Use one of the language codes from the IETF BCP 47 standard.",
+            "type": "string"
+        },
+        "license": {
+            "description": "The license for the resource. Use one of the SPDX license identifiers.",
+            "type": "object",
+            "properties": {
+                "id": {
+                    "type": "string"
+                },
+                "url": {
+                    "type": "string",
+                    "format": "uri"
+                }
+            }
+        },
+        "version": {
+            "description": "The version of the resource.",
+            "type": "string"
+        },
+        "references": {
```

Comment:
This is the main point of divergence between KBCMS and commonmeta, as we want to capture references as well as other types of relationship between resources (part of/has part; cites/cited by; etc.).

Reply:
In my view user interfaces are of course important, but the DataCite approach of using `related_identifiers` has created a situation that is confusing, not helping with the adoption of data citation. I think specifying a `references` property is one step towards making this simpler. We still need `related_identifiers` but they should focus on a few important relationships besides references, e.g. versioning and isPartOf/HasPart.

Reply:
And it makes sense to add the counterpart to relations, namely citations. For relation_types isCitedBy, isReferencedBy, isSupplementTo, see https://doi.org/10.5438/1843-k679.

Reply:
![references, citations, et al.](https://datacite.org/wp-content/uploads/2021/11/citations_display_fig1_.png)

I think ideally any user-facing interfaces for data entry would have references/related IDs/etc. divided as they are in this diagram, but behind the scenes in the database, things would be stored as related_identifiers and relationship types, with some sort of mapping file to interconvert between the UI and DB versions.

Reply:
I basically agree with this view but found that the different relation types have different requirements, and some level of `denormalization` in the database backend might help with performance and maintainability.

Reply:
Levels of confusion are not helped by DataCite having `related_identifiers` with relation type `isIdenticalTo` as well as `alternate_identifiers`!